config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +106,232 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerMainConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// +union
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Deployed and NotDeployed
```

Comment:
This ties a bit into my question about what the expected behavior is if the alertmanager config is unset by a user in the spec, but is NotDeployed the same as not specifying an alertmanager config?